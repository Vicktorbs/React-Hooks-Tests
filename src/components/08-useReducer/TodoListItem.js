import React from 'react'
import PropTypes from 'prop-types'

export const TodoListItem = ({todo, index, handleToggle, handleDelete}) => {
    return (
        <>
            <p className={ `${ todo.done && 'complete' }` } onClick={ () => handleToggle(todo.id) }>
                { index + 1 }. { todo.desc }
            </p>
            <button className="btn btn-danger" onClick={ () => { handleDelete(todo.id) } }>
                Borrar
            </button>
        </>
    )
}
TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired, 
    index: PropTypes.number.isRequired, 
    handleToggle: PropTypes.func.isRequired, 
    handleDelete: PropTypes.func.isRequired
}