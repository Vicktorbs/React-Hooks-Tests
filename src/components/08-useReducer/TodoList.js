import React from 'react'
import { TodoListItem } from './TodoListItem'
import PropTypes from 'prop-types'

export const TodoList = ({todos, handleDelete, handleToggle}) => {
    
    return (
        <ul className="list-group list-group-flush">
            {
                todos.map((todo, i) => (
                    <li className="list-group-item" key={ todo.id }>
                        <TodoListItem todo={ todo } index={ i } handleToggle={ handleToggle } handleDelete={ handleDelete } />
                        {/* <p className={ `${ todo.done && 'complete' }` } onClick={ () => handleToggle(todo.id) }>{ i + 1 }. { todo.desc }</p>
                        <button className="btn btn-danger" onClick={ () => { handleDelete(todo.id) } }>
                            Borrar
                        </button> */}
                    </li>
                ))
            }
        </ul>
    )
}
TodoList.propTypes = {
    todos: PropTypes.array.isRequired, 
    handleToggle: PropTypes.func.isRequired, 
    handleDelete: PropTypes.func.isRequired
}