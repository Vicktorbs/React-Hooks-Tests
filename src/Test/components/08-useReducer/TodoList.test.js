import React from 'react';
import { shallow } from 'enzyme';
import { TodoList } from '../../../components/08-useReducer/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Test on component <TodoList />', () => {

    const handleToggle = jest.fn();
    const handleDelete = jest.fn();

    const wrapper = shallow(<TodoList 
        todos={ demoTodos }
        handleDelete={ handleDelete }
        handleToggle={ handleToggle }
    />)
    
    test('should show itself correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should have two <TodoListItem />', () => {
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function))
    })
    
})
