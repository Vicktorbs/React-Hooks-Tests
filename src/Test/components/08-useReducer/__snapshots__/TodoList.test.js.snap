// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on component <TodoList /> should show itself correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <li
    className="list-group-item"
    key="1"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      todo={
        Object {
          "desc": "Learn React",
          "done": false,
          "id": 1,
        }
      }
    />
  </li>
  <li
    className="list-group-item"
    key="2"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      todo={
        Object {
          "desc": "Learn Mongo",
          "done": false,
          "id": 2,
        }
      }
    />
  </li>
</ul>
`;
